dispatch/dispatch.go

```diff
@@ -161,20 +167,44 @@ func (d *Dispatcher) run(it provider.AlertIterator) {
 				return
 			}
 
-			level.Debug(d.logger).Log("msg", "Received alert", "alert", alert)
+			// this block is wrapped in a function to make sure that the span
+			// is ended before the next alert is processed
+			func() {
+				traceCtx, span := tracer.Start(d.ctx, "dispatch.Dispatcher.handleAlert",
+					trace.WithAttributes(
+						attribute.String("alert.name", alert.Name()),
+						attribute.String("alert.fingerprint", alert.Fingerprint().String()),
+						attribute.String("alert.status", string(alert.Status())),
+						attribute.String("receiver", d.route.RouteOpts.Receiver),
+					),
+					// we'll use producer here since the alert is not processed
+					// synchronously
+					trace.WithSpanKind(trace.SpanKindProducer),
+				)
+				defer span.End()
+
+				// make a link to this span - we can't make the processAlert
```

Comment:
Do we need to be aware of constraints using links here? For example, the aggregation group `run` method can run for minutes/hours/days/weeks/years. What will happen if the linked trace is older than the retention period?